pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentRestletResource.java

```diff
@@ -230,19 +232,32 @@ protected Representation toggleSegmentState(String tableName, String segmentName
 
     JSONArray ret = new JSONArray();
     String tableNameWithType = null;
-    if ((tableType == null || TableType.OFFLINE.name().equalsIgnoreCase(tableType))
+    List<String> segmentsToToggle;
+
+    if (segmentName != null) {
+      segmentsToToggle = Collections.singletonList(segmentName);
+    } else {
+      segmentsToToggle = _pinotHelixResourceManager.getAllSegmentsForResource(tableName);
+    }
+    if ((tableType == null
+        && SegmentName.getSegmentType(segmentsToToggle.get(0)).equals(SegmentName.RealtimeSegmentType.UNSUPPORTED)
```

Comment:
The controller APIs take table name right now (the raw table name) and a table type independently. That said, it is good to add a check that user does not specify mytable_OFFLINE as table name (although, if they do, we will only come back with segment not found -- no harm done :)